lib/AIEFindFlows.cpp

```diff
@@ -176,26 +349,20 @@ class ConnectivityAnalysis {
       if (isa<FlowEndPoint>(other)) {
         // If we got to a tile, then add it to the result.
         connectedTiles.push_back(t);
-      } else if (auto switchOp = dyn_cast_or_null<SwitchboxOp>(other)) {
-        std::vector<PortMaskValue> nextPortMaskValues =
-            getConnectionsThroughSwitchbox(switchOp.connections(), otherPort);
-        std::vector<PacketConnection> newWorkList =
-            maskSwitchboxConnections(switchOp, nextPortMaskValues, maskValue);
-        // append to the worklist
-        worklist.insert(worklist.end(), newWorkList.begin(), newWorkList.end());
-        if (nextPortMaskValues.size() > 0 && newWorkList.size() == 0) {
-          // No rule matched some incoming packet.  This is likely a
-          // configuration error.
-          LLVM_DEBUG(llvm::dbgs() << "No rule matched incoming packet here: ");
-          LLVM_DEBUG(other->dump());
-        }
-      } else if (auto switchOp = dyn_cast_or_null<ShimMuxOp>(other)) {
+        connectedTilesIndex.push_back(
```

Comment:
So, here the code is creating a new graph data structure with the goal of giving good error messages.  If the data structure is only used to create error messages, maybe we should optimize for the 'there is no error case' and traverse the connections again if we find any antennas?  Only then would we create the graph?
We might still need some bookkeeping in order to detect other cases (like "dangling island" paths that don't start at a FlowEndPoint), I guess?